signature/jws/spec.go

```diff
@@ -47,3 +52,62 @@ func packPayload(desc notation.Descriptor, opts notation.SignOptions) *payload {
 		},
 	}
 }
+
+func checkCertChain(certChain []*x509.Certificate) error {
+	if len(certChain) == 0 {
+		return nil
+	}
+	if err := verifyCert(certChain[0], x509.ExtKeyUsageCodeSigning); err != nil {
+		return fmt.Errorf("signing certificate does not meet the minimum requirements: %w", err)
+	}
+	for _, c := range certChain[1:] {
+		for _, ext := range c.ExtKeyUsage {
+			if ext == x509.ExtKeyUsageTimeStamping {
+				if err := verifyCert(c, x509.ExtKeyUsageTimeStamping); err != nil {
+					return fmt.Errorf("timestamping certificate does not meet the minimum requirements: %w", err)
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// validateCert checks cert meets the requirements defined in
+// https://github.com/notaryproject/notaryproject/blob/main/signature-specification.md#certificate-requirements.
+func verifyCert(cert *x509.Certificate, extKeyUsage x509.ExtKeyUsage) error {
+	if cert.KeyUsage&x509.KeyUsageDigitalSignature == 0 {
+		return errors.New("keyUsage must have the bit positions for digitalSignature set")
+	}
+	var hasExtKeyUsage bool
+	for _, ext := range cert.ExtKeyUsage {
+		if ext == extKeyUsage {
+			hasExtKeyUsage = true
+			break
+		}
+	}
+	if !hasExtKeyUsage {
+		return fmt.Errorf("extKeyUsage must contain be %d", extKeyUsage)
+	}
+	for _, ext := range cert.Extensions {
+		switch ext.Id[3] {
```

Comment:
Are we guaranteed there are at least 4 elements here? Could not find that this was guaranteed looking through the docs.

Reply:
Good catch. Looking at this again I've noticed that I'm missing some more `Id` checks and unit tests. Will fix it.